custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -417,36 +417,35 @@ def get_payload(self, repeat_record, episode_case):
         }
 
     def handle_success(self, response, payload_doc, repeat_record):
-        # A successful response returns a Nikshay ID like 00001
-        # Failures also return with status code 200 and some message like
-        # Dublicate Entry or Invalid data format
-        # (Dublicate is not a typo)
         message = parse_SOAP_response(
             repeat_record.repeater.url,
             repeat_record.repeater.operation,
-            response
+            response,
         )
         try:
-            if isinstance(message, basestring) and message.isdigit():
-                health_facility_id = self._get_person_locations(payload_doc).pcp
-                nikshay_id = '-'.join([health_facility_id, message])
-                update_case(
-                    payload_doc.domain,
-                    payload_doc.case_id,
-                    {
-                        "private_nikshay_registered": "true",
-                        "nikshay_id": nikshay_id,
-                        "private_nikshay_error": "",
-                    },
-                    external_id=nikshay_id,
-                )
-            else:
-                self.handle_failure(message, payload_doc, repeat_record)
-        except NikshayResponseException as e:
-            _save_error_message(payload_doc.domain, payload_doc.case_id, unicode(e.message))
+            health_facility_id = self._get_person_locations(payload_doc).pcp
+        except NikshayLocationNotFound as e:
+            self.handle_exception(e, repeat_record)
+        else:
+            nikshay_id = '-'.join([health_facility_id, message])
+            update_case(
+                payload_doc.domain,
+                payload_doc.case_id,
+                {
+                    "private_nikshay_registered": "true",
+                    "nikshay_id": nikshay_id,
+                    "private_nikshay_error": "",
+                },
+                external_id=nikshay_id,
+            )
 
     def handle_failure(self, response, payload_doc, repeat_record):
-        _save_error_message(payload_doc.domain, payload_doc.case_id, unicode(response),
+        message = parse_SOAP_response(
```

Comment:
Also don't see why we are passing the response instead of just the message since we have parsed the response already in the repeater? Since we are storing just the final message anyway now.